src/database-storage/db-storage-interface.ts

```diff
@@ -0,0 +1,83 @@
+import {RawPdbDatabase, RawPrcDatabase} from 'palm-pdb';
+import {DlpReadUserInfoRespType} from '../protocols/dlp-commands';
+
+export interface DatabaseStorageInterface {
+  /**
+   * Creates the file system structure to hold database data for the username provided, may also
+   * create identification files regarding the specific PDA used.
+   * @param requestedUserName the username of the PDA
+   */
+  createUsernameInStorage(requestedUserName: string): Promise<void>;
+  /**
+   * Returns `true` if the requestedUserName was used in the `createUsernameInStorage()` function
+   *  in the past and may have data in this computer.
+   * @param requestedUserName the username of the PDA
+   */
+  isUsernameKnownInStorage(requestedUserName: string): Promise<boolean>;
```

Comment:
Maybe `userExists` instead of `known` to match `databaseExists` below?

Reply:
Hmm that makes a lot of sense, yes. Done.